pkg/catalog/manager/catalog_sync.go

```diff
@@ -43,13 +47,54 @@ func (m *Manager) Sync(key string, obj *v3.Catalog) (runtime.Object, error) {
 	}
 	logrus.Debugf("Chart hash comparison for global catalog %v: new -- %v --- current -- %v", catalog.Name, commit, catalog.Status.Commit)
 
+	// ensure the system catalog image cache exists
+	var systemCatalogImageCache *v1.ConfigMap
+	if catalog.Name == utils.SystemLibraryName {
+		systemCatalogImageCacheName := utils.GetCatalogImageCacheName(catalog.Name)
+		systemCatalogImageCache, err = m.ConfigMapLister.Get(systemCatalogImageCacheName, metav1.GetOptions{})
+		if err != nil {
+			if apiErr, ok := err.(*errors.StatusError); !ok || apiErr.Status().Code != 404 {
+				return nil, err
+			}
+		}
+
+		// if the cache does not exist generate it
+		if systemCatalogImageCache.UID == "" {
+			systemCatalogImageCache = &v1.ConfigMap{}
+			systemCatalogImageCache.Name = systemCatalogImageCacheName
+			err = image.CreateCatalogImageListConfigMap(systemCatalogImageCache, catalog)
+			if err != nil {
+				return nil, err
+			}
+
+			systemCatalogImageCache, err = m.ConfigMapLister.Create(systemCatalogImageCache)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
 	if isUpToDate(commit, catalog) {
 		if setRefreshed(catalog) {
 			m.catalogClient.Update(catalog)
 		}
 		return nil, nil
 	}
 
+	// update the system catalog image cache if it is out of date
+	if catalog.Name == utils.SystemLibraryName {
```

Comment:
You should combine this logic into the previous if block. So 
1. Check if configmap exists, if not, create configmap with catalog image list.
2. If configmap exists, check if catalog is update to date. If up to date, do nothing. Otherwise retrieve catalog image list and update the configmap.